core/tests/ts-integration/tests/l2-erc20.test.ts

```diff
@@ -0,0 +1,338 @@
+        await withdrawalTx.waitFinalize();
+
+        await sleep(60000);
+
+        await alice.finalizeWithdrawalParams(withdrawalTx.hash); // kl todo finalize the Withdrawals with the params here. Alternatively do in the SDK.
+        await expect(alice.finalizeWithdrawal(withdrawalTx.hash)).toBeAccepted();
+
+        tokenDetails.l1Address = await l1NativeTokenVault.tokenAddress(zkTokenAssetId);
+        const balanceAfterBridging = await alice.getBalanceL1(tokenDetails.l1Address);
+        expect(balanceAfterBridging).toEqual(10n);
+    });
+
+    test('Token properties are correct', async () => {
+        await expect(aliceErc20.name()).resolves.toBe(tokenDetails.name);
+        await expect(aliceErc20.decimals()).resolves.toBe(tokenDetails.decimals);
+        await expect(aliceErc20.symbol()).resolves.toBe(tokenDetails.symbol);
+        await expect(aliceErc20.balanceOf(alice.address)).resolves.toBeGreaterThan(0n); // 'Alice should have non-zero balance'
+    });
+
+    test('Can perform a transfer', async () => {
+        const value = 200n;
+
+        const balanceChange = await shouldChangeTokenBalances(tokenDetails.l2Address, [
+            { wallet: alice, change: -value },
+            { wallet: bob, change: value }
+        ]);
+        const feeCheck = await shouldOnlyTakeFee(alice);
+
+        // Send transfer, it should succeed.
+        await expect(aliceErc20.transfer(bob.address, value)).toBeAccepted([balanceChange, feeCheck]);
+    });
+
+    test('Can perform a transfer to self', async () => {
+        const value = 200n;
+
+        // When transferring to self, balance should not change.
+        const balanceChange = await shouldChangeTokenBalances(tokenDetails.l2Address, [{ wallet: alice, change: 0n }]);
+        const feeCheck = await shouldOnlyTakeFee(alice);
+        await expect(aliceErc20.transfer(alice.address, value)).toBeAccepted([balanceChange, feeCheck]);
+    });
+
+    test('Incorrect transfer should revert', async () => {
+        const value = ethers.parseEther('1000000.0');
+        const gasPrice = await scaledGasPrice(alice);
+
+        // Since gas estimation is expected to fail, we request gas limit for similar non-failing tx.
+        const gasLimit = await aliceErc20.transfer.estimateGas(bob.address, 1);
+
+        // Balances should not change for this token.
+        const noBalanceChange = await shouldChangeTokenBalances(tokenDetails.l2Address, [
+            { wallet: alice, change: 0n },
+            { wallet: bob, change: 0n }
+        ]);
+        // Fee in ETH should be taken though.
+        const feeTaken = await shouldOnlyTakeFee(alice);
+
+        // Send transfer, it should revert due to lack of balance.
+        await expect(aliceErc20.transfer(bob.address, value, { gasLimit, gasPrice })).toBeReverted([
+            noBalanceChange,
+            feeTaken
+        ]);
+    });
+
+    test('Transfer to zero address should revert', async () => {
```

Comment:
I think all these things like transfers etc are already checked in `erc20.test.ts`. Here in this test suite we only need to check withdrawal / deposit related stuff